tools/shared.py

```diff
@@ -1903,9 +1903,14 @@ def link_lld(args, target, opts=[], lto_level=0):
     if Settings.LINKABLE:
       cmd.append('--export-all')
 
+    # in standalone mode, crt1 will call the constructors from inside the wasm
+    if not Settings.STANDALONE_WASM:
```

Comment:
Except when there is no entry point we do want to export it.  Perhaps this should be guarded by `use_start_function` below?

Also might as will do this on one line: `cmd += [  '--export', '__wasm_call_ctors']`

Reply:
I'm unclear on what "no entry point" means. When does that happen?

Reply:
(and do we not have tests for it?)

Reply:
OK I think you right.   I though we were avoiding the _start function when building a STANDALONE_WASM without main, but actaully we still link _start in that case, its just weakly linked to main so it OK for main to be missing.

This change lgtm then, with cleanup